proxy/envoy/config_test.go

```diff
@@ -274,6 +275,36 @@ func testConfig(r *model.IstioRegistry, instance, envoyConfig string, t *testing
 	compareJSON(envoyConfig, t)
 }
 
+func testConfigWithSSLContext(r *model.IstioRegistry, instance, envoyConfig string, t *testing.T) {
+        ds := mock.Discovery
+        meshConfigWithSSLContext := &MeshConfig{
+                DiscoveryAddress: DefaultMeshConfig.DiscoveryAddress,
+                MixerAddress:     DefaultMeshConfig.MixerAddress,
+                ProxyPort:        DefaultMeshConfig.ProxyPort,
+                AdminPort:        DefaultMeshConfig.AdminPort,
+                BinaryPath:       DefaultMeshConfig.BinaryPath,
+                ConfigPath:       DefaultMeshConfig.ConfigPath,
+                EnableAuth:       true,
```

Comment:
copy the struct and update the field. you can stack allocate a pointer and that creates a copy.